src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericValueSetFactory.cs

```diff
@@ -16,56 +16,53 @@ internal static partial class ValueSetFactory
         /// parameterized by a type class
         /// <see cref="INumericTC{T}"/> that provides the primitives for that type.
         /// </summary>
-        private sealed class NumericValueSetFactory<T, TTC> : IValueSetFactory<T> where TTC : struct, INumericTC<T>
+        private struct NumericValueSetFactory<T, TTC> : IValueSetFactory<T> where TTC : class, INumericTC<T>
```

Comment:
>struct [](http://example.com/codeflow?start=16&length=6)

:memo: changed this to a struct as the places that box it already cache the boxed result and there's a couple of places that needed to allocate new ones (so a struct reduces allocations)